jstests/core/profile_sampling.js

```diff
@@ -0,0 +1,48 @@
+// special db for running tests in parallel
+var stddb = db;
+var db = db.getSisterDB("profile_sampling");
+var username = "jstests_profile_sampling_user";
+
+var t = db.profile_sampling;
+
+db.dropUser(username);
+db.dropDatabase();
+
+try {
+    db.createUser({user: username, pwd: 'password', roles: jsTest.basicUserRoles});
+    db.auth(username, 'password');
+
+    db.setProfilingLevel(0);
+    db.system.profile.drop();
+    assert.eq(0, db.system.profile.count());
+
+    db.createCollection(t.getName());
+    t.insert({x: 1});
+
+    assert.commandWorked(db.runCommand({profile: 2, sampleRate: 0}));
```

Comment:
Good question. Both seem reasonable. I'll let you make the call on which you think seems best. I'm not a huge fan of arguments that could mean two things, but the advantage is that it prevents users from having to do something like `db.setProfilingLevel(2, undefined, {sampleRate: 0.5});`.